File: GameRanks/src/main/java/GameRanks/GameRanks/serverStruct/StatisticStruct.java
package GameRanks.GameRanks.serverStruct;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class StatisticStruct {
    Iterable<Object[]> numberOfGamesPerGenre;
    Iterable<Object[]> numberOfGamesPerPlatform;
    Iterable<Object[]> numberOfReviewsPerPlatform;
    Iterable<Object[]> avgScorePerPlatform;
}

File: GameRanksClient/src/app/models/serverStructs/element/StatisticStruct.ts
export class StatisticStruct{
	numberOfGamesPerGenre: Object[][];
	numberOfGamesPerPlatform: Object[][];
	numberOfReviewsPerPlatform: Object[][];
	avgScorePerPlatform: Object[][];

	constructor(numberOfGamesPerGenre: Object[][], numberOfGamesPerPlatform: Object[][], numberOfReviewsPerPlatform: Object[][], avgScorePerPlatform: Object[][]){
		this.numberOfGamesPerGenre = numberOfGamesPerGenre;
		this.numberOfGamesPerPlatform = numberOfGamesPerPlatform;
		this.numberOfReviewsPerPlatform = numberOfReviewsPerPlatform;
		this.avgScorePerPlatform = avgScorePerPlatform;
	}
}
